monitoring/bin/create_elasticsearch_datasource.sh

```diff
@@ -0,0 +1,203 @@
+#! /bin/bash
+
+cd "$(dirname $BASH_SOURCE)/../.."
+source logging/bin/common.sh
+source monitoring/bin/common.sh
+
+source logging/bin/apiaccess-include.sh
+source logging/bin/secrets-include.sh
+source logging/bin/rbac-include.sh
+
+this_script=`basename "$0"`
+
+function show_usage {
+  log_message  "Usage: $this_script --namespace NAMESPACE --tenant TENANT --user USER --password PASSWORD"
+  log_message  ""
+  log_message  "'Creates the Elasticsearch datasource for a Viya tenant given using user-provided "
+  log_message  "namespace, tenant, and user credentials."
+  log_message  ""
+  log_message  "    Arguments:"
+  log_message  "     -ns, --namespace NAMESPACE   - The namespace where the Viya tenant is resides."
+  log_message  "     -t,  --tenant TENANT         - The tenant whose Elasticsearch data source you want to set up."
+  log_message  ""
+}
+
+# Assigning passed in parameters as variables for the script:
+POS_PARMS=""
+
+# Setting passed in variables:
+while (( "$#" )); do
+  case "$1" in
+    -ns|--namespace)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenantNS=$2
+        shift 2
+      else
+        log_error "A value for parameter [NAMESPACE] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check for parameters - set with cluster or tenant
+if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
+   cluster="true"
+elif [ -n "$tenant" ]; then
+    nst="${tenantNS}_${tenant}"
+else
+   nst="$tenantNS"
+fi
+
```

Comment:
So we are supporting the creation of a datasource scoped to a single namespace (without tenant)?  I am NOT opposed to that and think it could be useful...but don't recall talking about that.